Lib/test/test_imp.py

```diff
@@ -310,8 +310,12 @@ def test_multiple_calls_to_get_data(self):
         loader.get_data(imp.__file__)  # Will need to create a newly opened file
 
     def test_load_source(self):
-        with self.assertRaisesRegex(ValueError, 'embedded null'):
-            imp.load_source(__name__, __file__ + "\0")
+        # Create a temporary module since load_source() modifies
+        # sys.modules[name] attributes like __loader___
+        tmp_mod = type(sys.modules[__name__])(__name__)
+        with support.swap_item(sys.modules, __name__, tmp_mod):
+            with self.assertRaisesRegex(ValueError, 'embedded null'):
+                imp.load_source(__name__, __file__ + "\0")
```

Comment:
Wouldn't be safe to not use a name of existing module?

Reply:
Done.